lib/resin-lint.ts

```diff
@@ -5,6 +5,16 @@ import * as glob from 'glob';
 import * as optimist from 'optimist';
 import * as path from 'path';
 import * as tslint from 'tslint';
+import { promisify } from 'util';
+
+// TODO: Switch to using fs.promises when dropping node 8 support
+const realpathAsync = promisify(fs.realpath);
+const readFileAsync = promisify(fs.readFile);
+const existsAsync = promisify(fs.exists);
```

Comment:
`promisify(fs.exists)` probably doesn't work as intended, because:

> https://nodejs.org/docs/latest-v8.x/api/util.html#util_util_promisify_original
> `util.promisify(original)`
> Takes a function following the common error-first callback style, i.e. taking a (err, value) => ... callback as the last argument, and returns a version that returns promises.

But:
> https://nodejs.org/docs/latest-v8.x/api/fs.html#fs_fs_exists_path_callback
> `fs.exists(path, callback)`
> Stability: 0 - Deprecated: Use fs.stat() or fs.access() instead.
> **Note that the parameter to this callback is not consistent with other Node.js callbacks.** Normally, the first parameter to a Node.js callback is an err parameter, optionally followed by other parameters. The fs.exists() callback has only one boolean parameter. This is one reason fs.access() is recommended instead of fs.exists().

Reply:
Fwiw it'll work fine because `util.promisify` will use the custom promisified version: https://github.com/nodejs/node/blob/v8.17.0/lib/fs.js#L316-L322

Reply:
Switched to `fs.access`